datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -2560,6 +2562,101 @@ pub fn array_distinct(args: &[ArrayRef]) -> Result<ArrayRef> {
     }
 }
 
+pub fn array_resize(arg: &[ArrayRef]) -> Result<ArrayRef> {
+    if arg.len() < 2 || arg.len() > 3 {
+        return exec_err!("array_resize needs two or three arguments");
+    }
+
+    let new_len = as_int64_array(&arg[1])?;
+    let new_element = if arg.len() == 3 {
+        Some(arg[2].clone())
+    } else {
+        None
+    };
+
+    match &arg[0].data_type() {
+        DataType::List(field) => {
+            let array = as_list_array(&arg[0])?;
+            general_list_resize::<i32>(array, new_len, field, new_element)
+        }
+        DataType::LargeList(field) => {
+            let array = as_large_list_array(&arg[0])?;
+            general_list_resize::<i64>(array, new_len, field, new_element)
+        }
+        array_type => exec_err!("array_resize does not support type '{array_type:?}'."),
+    }
+}
+
+fn general_list_resize<O: OffsetSizeTrait>(
+    array: &GenericListArray<O>,
+    count_array: &Int64Array,
+    field: &FieldRef,
+    default_element: Option<ArrayRef>,
+) -> Result<ArrayRef> {
+    let mut offsets = vec![O::usize_as(0)];
+    let mut new_arrays = vec![];
+
+    let dt = array.value_type();
+    let converter = RowConverter::new(vec![SortField::new(dt.clone())])?;
+    let default_element = if let Some(default_element) = default_element {
```

Comment:
I think if you used ScalarValue here (rather than Option<ArrayRef>) you can probably avoid empty_list entirely